gui/src/app/FileEditor/TextEditor.tsx

```diff
@@ -186,39 +173,95 @@ const toMonacoMarkerSeverity = (
   }
 };
 
+type ToolbarProps = {
+  items: ToolbarItem[];
+  label: string;
+  onSaveText: () => void;
+  edited: boolean;
+  readOnly: boolean;
+};
+
+const ToolBar: FunctionComponent<ToolbarProps> = ({
+  items,
+  label,
+  onSaveText,
+  edited,
+  readOnly,
+}) => {
+  const toolBarItems = useMemo(() => {
+    const editorItems: ToolbarItem[] = [];
+
+    if (!readOnly && edited) {
+      editorItems.push({
+        type: "button",
+        icon: <Save />,
+        onClick: onSaveText,
+        tooltip: "Save file",
+        label: "Save",
+      });
+    }
+
+    if (edited) {
+      editorItems.push({
+        type: "text",
+        label: "Edited",
+        color: "red",
+      });
+    }
+
+    if (readOnly) {
+      editorItems.push({
+        type: "text",
+        label: "Read Only",
+        color: "gray",
+      });
+    }
```

Comment:
Since the `readOnly` is logically incompatible with `edited`, this could be two simple conditionals:

```
if (readOnly) {
  push read only
} else {
  if (edited) {
    push edited stuff
  }
}
```